pkg/types/azure/machinepool.go

```diff
@@ -38,4 +44,18 @@ func (a *MachinePool) Set(required *MachinePool) {
 	if required.OSDisk.DiskSizeGB != 0 {
 		a.OSDisk.DiskSizeGB = required.OSDisk.DiskSizeGB
 	}
+
+	a.OSDisk.DiskType = validateDiskType(required.OSDisk.DiskType)
+}
+func validateDiskType(diskType string) string {
```

Comment:
validating and setting defaults are two different functions

Reply:
Couldn't find an appropriate function name for this. It validates the value and sets the default when there is empty or invalid value.